skupper-router.spec.rpkg

```diff
@@ -0,0 +1,144 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# This .spec file uses `rpkg` (https://pagure.io/rpkg-util) to provide
+# pleasant user experience to developers.
+#
+#  Command                  Description
+# `rpkg srpm`              Creates a *.src.rpm file in /tmp/rpkg (exact path is printed)
+# `rpkg local --nocheck`   Builds a *.rpm for your system in /tmp/rpkg (exact path is printed)
+#
+# See `man rpkg` for more commands.
+
+%global proton_minimum_version 0.34.0
+%global libwebsockets_minimum_version 3.0.1
+%global libnghttp2_minimum_version 1.33.0
+
+Name:          skupper-router
+Version:       {{{ git_dir_version }}}
+Release:       2.0.0%{?dist}
+Summary:       Router for Skupper.io
+License:       ASL 2.0
+URL:           https://skupper.io
+VCS:           {{{ git_dir_vcs }}}
+
+BuildRequires: cmake
+BuildRequires: python3-devel
+BuildRequires: openssl-devel
+BuildRequires: qpid-proton-c-devel >= %{proton_minimum_version}
+BuildRequires: libwebsockets-devel >= %{libwebsockets_minimum_version}
+BuildRequires: libnghttp2-devel >= %{libnghttp2_minimum_version}
+# man pages --help
+BuildRequires: asciidoc
+BuildRequires: python3-qpid-proton >= %{proton_minimum_version}
+
+Source: {{{ git_dir_pack }}}
+
+%description
+A lightweight message router, written in C and built on Qpid Proton, that provides flexible and scalable interconnect backend for Skupper.io Level 7 Virtual Application Network.
+
+%prep
+{{{ git_dir_setup_macro }}}
+
+%build
+%cmake -DUSE_LIBWEBSOCKETS=ON -DUSE_LIBNGHTTP2=ON -DPython_EXECUTABLE=%{python3}
+%cmake_build --target all --target man
+
+%install
+%cmake_install
+
+%check
+%ctest
+
+# TODO: name?
+%package server
+Summary:  The skrouterd deamon
+Requires: python3
+Requires: qpid-proton-c >= %{proton_minimum_version}
+Requires: libwebsockets >= %{libwebsockets_minimum_version}
+Requires: libnghttp2 >= %{libnghttp2_minimum_version}
+
+%description server
+%{summary}.
+
+%files server
+/usr/sbin/skrouterd
+
+%config /etc/skupper-router/skrouterd.conf
+%config /etc/sasl2/skrouterd.conf
+
+%{python3_sitelib}/skupper_router/
+%{python3_sitelib}/skupper_router_site.py
+%{python3_sitelib}/__pycache__/skupper_router_site.*.pyc
+%{python3_sitelib}/skupper_router-*.egg-info
+
+# TODO: skupper-router-tools depend on tools/command.py, etc. in skupper_router_internal/
+/usr/lib/skupper-router/python/skupper_router_internal/
```

Comment:
The qpid-dispatch packages have the same problem. It is permitted to install `qpid-dispatch-tools` without `qpid-dispatch-router`, but then qdmanage/qdstat will fail with

```
$ qdmanage
Traceback (most recent call last):
  File "/usr/bin/qdmanage", line 27, in <module>
    import qpid_dispatch_site  # noqa: F401 # imported for side-effects (inserts to sys.path)
ModuleNotFoundError: No module named 'qpid_dispatch_site'
```

Reply:
@kgiusti @mcressman This is a bug, ^^^, I'd say. I am not sure how to best resolve this. Either make the `-tools` package depend on the server package, or split `qpid-dispatch-site` between tools and the router. The dependency is weird, but the server package is relatively small, and trying to do the python module split is somehow even weirder.

Reply:
(Packaging `qpid-dispatch-site` in a separate package that both -server and -tools depend on is certainly also possible, but that would result in a lot of packages)

Reply:
I'll go on-record and say we should split out the python bits that are used for the client tools from the router-internal python code.  Ideally the client python stuff could be a stand-alone python package (available via pypi) that the tools/tests/etc would depend upon.

Not advocating that split should be done as part of this patch of course, but a separate future enhancement for some plucky python developer to run with :) ...

Reply:
do new rpm package, call it `-common`

Reply:
done